org.gnome.NautilusPreviewer.json

```diff
@@ -0,0 +1,117 @@
+{
+    "app-id": "org.gnome.NautilusPreviewer",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "41",
+    "sdk": "org.gnome.Sdk",
+    "build-options": {
+        "cflags": "-O2 -g",
+        "cxxflags": "-O2 -g",
+        "env": {
+            "V": "1"
+        }
+    },
+    "command": "sushi",
+    "cleanup": [
+        "/include",
+        "/lib/pkgconfig",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.a"
+    ],
+    "finish-args": [
+        "--socket=pulseaudio",
+        "--socket=x11",
+        "--socket=wayland",
+        "--share=ipc",
+        "--share=network",
+        "--device=dri",
+        "--talk-name=org.freedesktop.FileManager1",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--filesystem=host:ro",
```

Comment:
does it really need host?

Reply:
it can be invoked on any file that nautilus can see and has to work seamlessly without any kind of user confirmation so yes, it does.

Reply:
if it can work `:ro` when why isn't home `:ro` too? Does it write stuff in HOME that it doesn't elswhere?

Reply:
I've removed the permission to home directory and it seems to work fine.

What never worked on my computer (Fedora 35 Silverblue, btrfs partition) is the preview on any file outside of the home directory. I don't think it's a sandbox issue, as I tried to grant all host permissions temporarily and it didn't help. In the log I see the error: `org.gnome.Nauti[50136]: JS ERROR: Failed to query default handler for file:///var/www/html/index.html: Gio.IOErrorEnum: Error when getting information for file “/var/www/html/index.html”: No such file or directory`.

Reply:
Basically, those files aren't mounted there when seen from inside the sandbox - they're available but at a different path. Sushi will need to be able to detect being inside a flatpak and then do some path translation for that case. Check the docs but basically you need that at /var/run/var etc. https://docs.flatpak.org/en/latest/sandbox-permissions.html?highlight=host#filesystem-access

Reply:
I don't think `/var` here is accessible from the sandbox.

Reply:
Can I leave this to the user?

I guess that most people will use Sushi to preview files in the home directory.
Those who need access to other directories can add a permission manually.

Reply:
Eventually I decided to follow what Felipe did on [upstream MR 27](https://gitlab.gnome.org/GNOME/sushi/-/merge_requests/27) and use `--filesystem=home:ro` only.